src/main/java/com/hotels/hcommon/hive/metastore/paths/PartitionedTablePathResolver.java

```diff
@@ -0,0 +1,98 @@
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.metastore.IMetaStoreClient;
+import org.apache.hadoop.hive.metastore.Warehouse;
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hadoop.hive.metastore.api.NoSuchObjectException;
+import org.apache.hadoop.hive.metastore.api.Partition;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.thrift.TException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.hotels.hcommon.hive.metastore.iterator.PartitionIterator;
+
+class PartitionedTablePathResolver implements TablePathResolver {
+
+  private static final Logger log = LoggerFactory.getLogger(PartitionedTablePathResolver.class);
+
+  private final Path tableBaseLocation;
+  private final Path globPath;
+  private final IMetaStoreClient metaStore;
+  private final Table table;
+
+  PartitionedTablePathResolver(IMetaStoreClient metaStore, Table table)
+      throws NoSuchObjectException, MetaException, TException {
+    this.metaStore = metaStore;
+    this.table = table;
+    log.debug("Table '{}' is partitioned", Warehouse.getQualifiedName(table));
+    tableBaseLocation = locationAsPath(table);
+    List<Partition> onePartition = metaStore.listPartitions(table.getDbName(), table.getTableName(), (short) 1);
+    if (onePartition.isEmpty()) {
+      String errorMessage = String.format("Table '%s' has no partitions, perhaps you can simply delete: %s.",
+          Warehouse.getQualifiedName(table), tableBaseLocation);
+      log.warn(errorMessage);
+      throw new RuntimeException(errorMessage);
+    }
+    Path partitionLocation = locationAsPath(onePartition.get(0));
+    int branches = partitionLocation.depth() - tableBaseLocation.depth();
+    String globSuffix = StringUtils.repeat("*", "/", branches);
+    globPath = new Path(tableBaseLocation, globSuffix);
+  }
+
+  @Override
+  public Path getGlobPath() {
+    return globPath;
+  }
+
+  @Override
+  public Path getTableBaseLocation() {
+    return tableBaseLocation;
+  }
+
+  @Override
+  public Set<Path> getMetastorePaths(short batchSize) throws NoSuchObjectException, MetaException, TException {
+    Set<Path> metaStorePaths = new HashSet<>();
+    PartitionIterator partitionIterator = new PartitionIterator(metaStore, table, batchSize);
+    while (partitionIterator.hasNext()) {
+      Partition partition = partitionIterator.next();
+      Path location = PathUtils.normalise(locationAsPath(partition));
+      if (!location
+          .toString()
+          .toLowerCase(Locale.ROOT)
+          .startsWith(tableBaseLocation.toString().toLowerCase(Locale.ROOT))) {
+        String errorMessage = String.format("Check your configuration: '%s' does not appear to be part of '%s'.",
+            location, tableBaseLocation);
+        log.error(errorMessage);
```

Comment:
Similar here, we shouldn't log if we're going to throw since the exception should end up in the logs anyway.